imcp/icmp.go

```diff
@@ -53,3 +53,84 @@ func SendIMCP(localAddr string, dst net.Addr, ttl, pid int, timeout time.Duratio
 	hop.Success = true
 	return hop, err
 }
+
+// SendIMCP sends a IMCP to a given destination
+func SendIMCP(localAddr string, dst net.Addr, ttl, pid int, timeout time.Duration, seq int) (hop ICMPReturn, err error) {
+	hop.Success = false
+	start := time.Now()
+	c, err := icmp.ListenPacket("ip4:icmp", localAddr)
+	if err != nil {
+		return hop, err
+	}
+	defer c.Close()
+	c.IPv4PacketConn().SetTTL(ttl)
```

Comment:
Error return value of `(*github.com/tonobo/mtr/vendor/golang.org/x/net/ipv4.genericOpt).SetTTL` is not checked